calour/io.py

```diff
@@ -472,12 +477,82 @@ def read_open_ms(data_file, sample_metadata_file=None, gnps_file=None, feature_m
         # load the gnps table
         gnps_data = pd.read_table(gnps_file, sep='\t')
         exp.exp_metadata['_calour_metabolomics_gnps_table'] = gnps_data
+        # link each feature to the gnps ids based on MZ/RT
+        exp._prepare_gnps_ids()
         # add gnps names to the features
         exp._prepare_gnps()
 
     return exp
 
 
+def read_gnps_ms(data_file, sample_metadata_file=None, gnps_file=None, feature_metadata_file=None,
+                 description=None, sparse=False, *, normalize, **kwargs):
+    '''Load a GNPS metabolomics experiment (exported from GNPS).
+    Parameters
+    ----------
+    data_file : str
+        name of the OpenMS bucket table CSV file.
+    sample_metadata_file : str or None, optional
+        None (default) to not load metadata per sample
+        str to specify name of sample mapping file (tsv)
+    gnps_file : str or None, optional
+        name of the gnps clusterinfosummarygroup_attributes_withIDs_arbitraryattributes/XXX.tsv file
```

Comment:
missing separations between words

Reply:
this is actually the gnps filename we need :)
see [here](https://bix-lab.ucsd.edu/display/Public/Cytoscape+2.8.X+Visualization+and+Analysis+Documentation)